code/modules/reagents/Chemistry-Machinery.dm

```diff
@@ -101,7 +101,9 @@
 
 	var chemicals[0]
 	for (var/re in dispensable_reagents)
-		var/datum/reagent/temp = chemical_reagents_list[re]
+		var/datum/reagent/temp
+		if(re in chemical_reagents_list)
+			temp = chemical_reagents_list[re]

```

Comment:
I view it as additional safety checks. Even if I find where the bad call is being made, I'd still have the checks here in case someone miscalled the function in future.